bender:
  # (Only one instance per robot) Publish all joint states
  joint_state_controller:
    type: joint_state_controller/JointStateController
    publish_rate: 50

  # Left arm controller
  l_arm_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - l_shoulder_pitch_joint
      - l_shoulder_roll_joint
      - l_shoulder_yaw_joint
      - l_elbow_pitch_joint
      - l_elbow_yaw_joint
      - l_wrist_pitch_joint
    gains:
      l_shoulder_pitch_joint: {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      l_shoulder_roll_joint:  {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      l_shoulder_yaw_joint:   {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      l_elbow_pitch_joint:    {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      l_elbow_yaw_joint:      {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      l_wrist_pitch_joint:    {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}

    constraints:
      goal_time: 5.0
      stopped_velocity_tolerance: 1.0
      l_shoulder_pitch_joint: {trajectory: 0.2, goal: 0.2}
      l_shoulder_roll_joint: {trajectory: 0.2, goal: 0.2}
      l_shoulder_yaw_joint: {trajectory: 0.2, goal: 0.2}
      l_elbow_pitch_joint: {trajectory: 0.2, goal: 0.2}
      l_elbow_yaw_joint: {trajectory: 0.2, goal: 0.2}
      l_wrist_pitch_joint: {trajectory: 0.2, goal: 0.2}

    state_publish_rate:  25            # Override default
    action_monitor_rate: 30            # Override default
    stop_trajectory_duration: 0        # Override default

  # Right arm controller
  r_arm_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - r_shoulder_pitch_joint
      - r_shoulder_roll_joint
      - r_shoulder_yaw_joint
      - r_elbow_pitch_joint
      - r_elbow_yaw_joint
      - r_wrist_pitch_joint
    gains:
      r_shoulder_pitch_joint: {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      r_shoulder_roll_joint:  {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      r_shoulder_yaw_joint:   {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      r_elbow_pitch_joint:    {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      r_elbow_yaw_joint:      {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}
      r_wrist_pitch_joint:    {p: 100.0, i: 10.0, d: 0.5, i_clamp: 0.0}

    constraints:
      goal_time: 5.0
      stopped_velocity_tolerance: 1.0
      r_shoulder_pitch_joint: {trajectory: 0.2, goal: 0.2}
      r_shoulder_roll_joint: {trajectory: 0.2, goal: 0.2}
      r_shoulder_yaw_joint: {trajectory: 0.2, goal: 0.2}
      r_elbow_pitch_joint: {trajectory: 0.2, goal: 0.2}
      r_elbow_yaw_joint: {trajectory: 0.2, goal: 0.2}
      r_wrist_pitch_joint: {trajectory: 0.2, goal: 0.2}

    state_publish_rate:  25            # Override default
    action_monitor_rate: 30            # Override default
    stop_trajectory_duration: 0        # Override default

  # Right gripper controller controller
  r_gripper_controller:
    type: effort_controllers/MultiGripperActionController
    joints:
      - r_int_finger_joint
      - r_ext_finger_joint
    goal_tolerance: 0.05
    max_effort: 3.1 # RX28 at 15.5 V
    stall_velocity_threshold: 0.02
    stall_timeout: 0.2
    gains:
      r_int_finger_joint: {p: 100.0, i: 1.0, d: 1.0, i_clamp: 3.1}
      r_ext_finger_joint: {p: 100.0, i: 1.0, d: 1.0, i_clamp: 3.1}

  # Left gripper controller controller
  l_gripper_controller:
    type: effort_controllers/MultiGripperActionController
    joints:
      - l_int_finger_joint
      - l_ext_finger_joint
    goal_tolerance: 0.05
    max_effort: 3.1 # RX28 at 15.5 V
    stall_velocity_threshold: 0.02
    stall_timeout: 0.2
    gains:
      l_int_finger_joint: {p: 100.0, i: 1.0, d: 1.0, i_clamp: 3.1}
      l_ext_finger_joint: {p: 100.0, i: 1.0, d: 1.0, i_clamp: 3.1}

  # Head controller
  head_controller:
    type: effort_controllers/JointTrajectoryController
    joints:
      - head_pitch_joint
      - head_yaw_joint
    gains:
      head_pitch_joint:  {p: 100.0, i: 10.0, d: 0.5, i_clamp: 5.0}
      head_yaw_joint: {p: 100.0, i: 10.0, d: 0.5, i_clamp: 5.0}

    constraints:
      goal_time: 5.0
      stopped_velocity_tolerance: 1.0
      head_pitch_joint: {trajectory: 0.2, goal: 0.2}
      head_yaw_joint: {trajectory: 0.2, goal: 0.2}

    state_publish_rate:  25            # Override default
    action_monitor_rate: 30            # Override default
    stop_trajectory_duration: 0        # Override default